hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/virtual/entities/OnDiskAccount.java

```diff
@@ -0,0 +1,947 @@
+        System.arraycopy(that.longs, 0, this.longs, 0, LongValues.COUNT);
+    }
+
+    public static OnDiskAccount from(final MerkleAccountState inMemoryAccount) {
+        final var onDiskAccount = new OnDiskAccount();
+
+        // Objects
+        onDiskAccount.setMemo(inMemoryAccount.memo());
+        onDiskAccount.setAlias(inMemoryAccount.getAlias());
+        onDiskAccount.setHbarAllowances(inMemoryAccount.getCryptoAllowances());
+        onDiskAccount.setFungibleAllowances(inMemoryAccount.getFungibleTokenAllowances());
+        onDiskAccount.setNftOperatorApprovals(inMemoryAccount.getApproveForAllNftsUnsafe());
+        onDiskAccount.setKey(inMemoryAccount.key());
+        // Flags
+        onDiskAccount.setIsDeleted(inMemoryAccount.isDeleted());
+        onDiskAccount.setIsContract(inMemoryAccount.isSmartContract());
+        onDiskAccount.setIsDeclineReward(inMemoryAccount.isDeclineReward());
+        onDiskAccount.setIsReceiverSigRequired(inMemoryAccount.isReceiverSigRequired());
+        // Ints
+        onDiskAccount.setNumContractKvPairs(inMemoryAccount.getNumContractKvPairs());
+        onDiskAccount.setMaxAutoAssociations(inMemoryAccount.getMaxAutomaticAssociations());
+        onDiskAccount.setUsedAutoAssociations(inMemoryAccount.getUsedAutomaticAssociations());
+        onDiskAccount.setNumAssociations(inMemoryAccount.getNumAssociations());
+        onDiskAccount.setNumPositiveBalances(inMemoryAccount.getNumPositiveBalances());
+        onDiskAccount.setNumTreasuryTitles(inMemoryAccount.getNumTreasuryTitles());
+        // Longs
+        onDiskAccount.setExpiry(inMemoryAccount.expiry());
+        onDiskAccount.setHbarBalance(inMemoryAccount.balance());
+        onDiskAccount.setAutoRenewSecs(inMemoryAccount.autoRenewSecs());
+        onDiskAccount.setNftsOwned(inMemoryAccount.nftsOwned());
+        onDiskAccount.setAccountNumber(inMemoryAccount.number());
+        onDiskAccount.setHeadTokenId(inMemoryAccount.getHeadTokenId());
+        onDiskAccount.setHeadNftId(inMemoryAccount.getHeadNftId());
+        onDiskAccount.setHeadNftSerialNum(inMemoryAccount.getHeadNftSerialNum());
+        onDiskAccount.setEthereumNonce(inMemoryAccount.ethereumNonce());
+        onDiskAccount.setStakedToMe(inMemoryAccount.getStakedToMe());
+        onDiskAccount.setStakePeriodStart(inMemoryAccount.getStakePeriodStart());
+        onDiskAccount.setStakedNum(inMemoryAccount.getStakedNum());
+        onDiskAccount.setStakeAtStartOfLastRewardedPeriod(
+                inMemoryAccount.getStakeAtStartOfLastRewardedPeriod());
+        if (inMemoryAccount.hasAutoRenewAccount()) {
+            onDiskAccount.setAutoRenewAccountNumber(inMemoryAccount.getAutoRenewAccount().num());
+        }
+        // Complex
+        onDiskAccount.setFirstStorageKey(inMemoryAccount.getFirstUint256Key());
+
+        return onDiskAccount;
+    }
+
+    @Override
+    public OnDiskAccount copy() {
+        this.immutable = true;
+        return new OnDiskAccount(this);
```

Comment:
Am I understanding this right, that if anyone calls `copy()` on this instance, then we're never supposed to make a change to anything in this object (or its class field instances) again?